mmv1/products/iam3/PrincipalAccessBoundaryPolicy.yaml

```diff
@@ -0,0 +1,163 @@
+references:
+  guides:
+    'Create and apply Principal Access Boundaries': 'https://cloud.google.com/iam/docs/principal-access-boundary-policies-create'
+  api: 'https://cloud.google.com/iam/docs/reference/rest/v3beta/organizations.locations.principalAccessBoundaryPolicies'
+min_version: 'beta'
+id_format: 'organizations/{{organization}}/locations/{{location}}/principalAccessBoundaryPolicies/{{principal_access_boundary_policy_id}}'
+base_url: 'organizations/{{organization}}/locations/{{location}}/principalAccessBoundaryPolicies'
+self_link: 'organizations/{{organization}}/locations/{{location}}/principalAccessBoundaryPolicies/{{principal_access_boundary_policy_id}}'
+create_url: 'organizations/{{organization}}/locations/{{location}}/principalAccessBoundaryPolicies?principalAccessBoundaryPolicyId={{principal_access_boundary_policy_id}}'
+update_verb: 'PATCH'
+update_mask: true
+import_format:
+  - 'organizations/{{organization}}/locations/{{location}}/principalAccessBoundaryPolicies/{{principal_access_boundary_policy_id}}'
+timeouts:
+  insert_minutes: 20
+  update_minutes: 20
+  delete_minutes: 20
+autogen_async: true
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    path: 'name'
+    wait_ms: 1000
+  result:
+    path: 'response'
+    resource_inside_response: true
+  error:
+    path: 'error'
+    message: 'message'
+examples:
+  - name: 'iam3_principal_access_boundary_policy'
+    min_version: 'beta'
+    primary_resource_id: 'my-pab-policy'
+    test_env_vars:
+      org_id: 'ORG_ID'
+    vars:
+      organization: 'ORG_ID'
+      location: 'global'
+      display_name: 'test pab policy'
+      pab_id: 'test-pab-policy'
+parameters:
+  - name: 'organization'
+    type: String
+    description: |
+      The parent organization of the principal access boundary policy.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'location'
+    type: String
+    description: |
+      The location the principal access boundary policy is in.
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'principalAccessBoundaryPolicyId'
+    type: String
+    description: |
+      The ID to use to create the principal access boundary policy.
+      This value must start with a lowercase letter followed by up to 62 lowercase letters, numbers, hyphens, or dots. Pattern, /a-z{2,62}/.
+    url_param_only: true
+    required: true
+    immutable: true
+properties:
+  - name: 'name'
+    type: String
+    description: |
+      Identifier. The resource name of the principal access boundary policy.  The following format is supported:
+       `organizations/{organization_id}/locations/{location}/principalAccessBoundaryPolicies/{policy_id}`
+    output: true
+  - name: 'uid'
+    type: String
+    description: |
+      Output only. The globally unique ID of the principal access boundary policy.
+    output: true
+  - name: 'etag'
```

Comment:
I believe that the etag field would only ever be used internally in the provider, and a user wouldn't set this field in their Terraform configuration. Because of that, I think this field should be `output: true`, instead of `default_from_api: true`. What do you think?

Reply:
a user can set an etag when calling update so they can ensure they specify a specific resource to update. i added the default_from_api b/c i kept getting an error in the update test for not testing that field.

Reply:
here's the error from testing:

=== NAME  TestAccIAM3PrincipalAccessBoundaryPolicy_iam3PrincipalAccessBoundaryPolicyExample
    resource_iam3_principal_access_boundary_policy_generated_test.go:42: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_iam3_principal_access_boundary_policy.my-pab-policy will be updated in-place
          ~ resource "google_iam3_principal_access_boundary_policy" "my-pab-policy" {
              - etag                                = "W/\"0MJpU5Y4jPJAMvKXTwD1ig==\"" -> null

Reply:
Hmm, that error's interesting.

Is the etag used to prevent concurrent writes to the policy?

Reply:
it's moreso something to manage concurrent writes to the policy. by specifying the etag, the user can be absolutely sure of delta of the change.

Reply:
Indeed, then I think we'd need to follow the way etags are used elsewhere in the provider. For example [google_organization_policy](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_organization_policy#etag) uses an etag but it's not expected for the user to provide a new etag value each time there's an update. Similarly all IAM resources use etags ([example](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/api_gateway_api_iam#etag)) and they are managed entirely in the provider without user input.

Reply:
thanks Sarah. went ahead and followed those resources and set output: true.

Reply:
Thanks! In the debug logs from the acceptance test I can see that the etag is stored in state after it's received from the API, and in subsequent API calls to update the policy I can see the etag is sent with the update request. The new etag returned from the API is then stored in the resource state and the process repeats. When Terraform performs refresh actions, prior to planning/applying changes, the refreshes will pull in etag changes that may have happened out-of-band.